### DISCUSSION TITLE
Custom Alert Message

### DISCUSSION THREAD

**Discussion by sudheeshb** [TwiN/gatus#320] [Q&A] [answered]
### Describe the feature request

Hello,

As of now  I am getting a lots of messages for the alert.  For example, I am using ping, and I get the following messages  as part of the alert, which is too much for my taste.

__________________________________
"
Gatus

An alert for xxxx has been detected.
Heath check failed 1 times in a row,

Description:
Down

Condition results
X  CONNECTED = false == true.

"
___________________________________

Instead of the whole bunch of messages above, I just need one line lke "UP" or "DOWN".    This means I need only the description part as part of the alert.  Is it possible?

I know  I can customize the Description

### Why do you personally want this feature to be implemented?

Too much messages in telegram alert with out adding any values.  I know what I have already configured, and I don't need that again.  This causes Readability issues.

### How long have you been using this project?

2 months

**Comment by TwiN** [comment]
Have you tried using the custom alerting provider? You should be able to achieve what you're trying to do with it.

**Comment by sudheeshb** [comment]
Thanks!.   This should work.  Is there any example available for custom alert using Telegram?. I see only  examples using slack in the documentation.

**Comment by TwiN** [comment]
You can look at the code for Telegram's alerting provider.

For instance, this is what the body looks like: https://github.com/TwiN/gatus/blob/f01b66f083c3184762afd8041877f09aef2f05b3/alerting/provider/telegram/telegram.go#L77

The URL: https://github.com/TwiN/gatus/blob/f01b66f083c3184762afd8041877f09aef2f05b3/alerting/provider/telegram/telegram.go#L38

**Comment by sudheeshb** [comment]
Thanks for the pointers.   Will check the source code

**Comment by sudheeshb** [comment]
I added the following , but couldn't make it work. 


alerting:
  custom:
     placeholders:
       ALERT_TRIGGERED_OR_RESOLVED:
         TRIGGERED: "partial_outage"
         RESOLVED: "operational"
    url: "https://api.telegram.org/bot123456:xxx_OraQhshL15muFs4C10Axxxxxxxx"
    method: "POST"
    body: |
      {
        "text": "50x41xxxx,   [ALERT_TRIGGERED_OR_RESOLVED], MARKDOWN"
      }


Got the following error.

main.main()
	/app/main.go:19 +0x168
2022/08/15 05:21:14 [config][Load] Reading configuration from configFile=config/config.yaml
panic: yaml: line 18: did not find expected key
goroutine 1 [running]:


Appreciate any help.   I am not a web developer by profession.
I believe the body is wrong.  The alert works fine if I use builtin telegram.  Also I API key/chatID etc are correct as it works with builtin telegram plugin. ( I didn't provide the correct value in the above code )

**Comment by TwiN** [comment, marked as answer]
Could you format the configuration properly so that I can see exactly what your config is? It seems like you're having an issue with the indentation :) 

To keep the code formatted, simply wrap the code with \`\`\`, such as:

\`\`\`
code goes here
\`\`\`

Also, the body will look more like this:
```json
{"chat_id": "50x41xxxx", "text": "[ALERT_TRIGGERED_OR_RESOLVED]", "parse_mode": "MARKDOWN"}
```
and the url:
```
https://api.telegram.org/bot123456/sendMessage
```

**Reply by sudheeshb** [reply]
Thanks.   It was indeed an issue with indentation!.  Made some progress.  Got the following error now.  Alert is failing.  Looks like still some thing wrong with the body.  I updated the URL and body as per your suggestions.  Thanks for the help so far!

2022/08/16 05:37:58 [controller][Handle] Listening on 0.0.0.0:8080
2022/08/16 05:37:59 [watchdog][execute] Monitored group=external; endpoint=example; success=true; errors=0; duration=82ms
2022/08/16 05:38:10 [watchdog][execute] Monitored group=Phones; endpoint=Sudheesh; success=false; errors=0; duration=10s
2022/08/16 05:38:10 [watchdog][handleAlertsToTrigger] Sending custom alert because alert for endpoint=Sudheesh with description='Down' has been TRIGGERED
2022/08/16 05:38:10 [watchdog][handleAlertsToTrigger] Failed to send an alert for endpoint=Sudheesh: call to provider alert returned status code 400: {"ok":false,"error_code":400,"description":"Bad Request: message text is empty"}


```
> alerting:
>   custom:
>     placeholders:
>       ALERT_TRIGGERED_OR_RESOLVED:
>         TRIGGERED: "partial_outage"
>         RESOLVED: "operational"
>     url: "https://api.telegram.org/bot123456:CCC_xxxxxxxxxxxxxxx/sendMessage"
>     method: "POST"
>     body: |
>      {
>        "chat_id": "50xxxxxxxx", "text": "testing", "parse_mode": "MARKDOWN"
>      }
> 

```

**Reply by TwiN** [reply]
The body might still not be properly indented:
```yaml
alerting:
  custom:
    url: "https://api.telegram.org/bot123456:CCC_xxxxxxxxxxxxxxx/sendMessage"
    method: "POST"
    headers:
      Content-Type: "application/json"
    placeholders:
      ALERT_TRIGGERED_OR_RESOLVED:
        TRIGGERED: "partial_outage"
        RESOLVED: "operational"
    body: |
      {
        "chat_id": "50xxxxxxxx", "text": "testing", "parse_mode": "MARKDOWN"
      }

```
I also added the Content-Type header for good measure, but it shouldn't be necessary.


If you want to be on the safe side, you could always just put the body in one line. 
Something like this should work:
```
    body: "{'chat_id': '50xxxxxxxx', 'text': 'testing', 'parse_mode': 'MARKDOWN'}"
```

**Comment by sudheeshb** [comment]
Works fine now!.  Thanks a lot.  I just copied the above code , updated the keys/chat_id.  Kept the body as it is.   All good now

**Comment by sudheeshb** [comment]
I noticed that the time zone used in the 'response time'  graph of the end points doesn't match with local time zone.  What am I missing here?

The time zone in the first page ( health status ) is correct.  I am using Docker and passing the time stamp via   '-e TZ=America/Los_Angeles' .  

docker run -d \
  -p 9090:8080 \
  -v /home/testDocker/Gatus/config:/config \
  --name gatus \
  -e TZ=America/Los_Angeles \
  --dns 192.168.1.1 \
  --restart=unless-stopped \
 twinproduction/gatus

**Reply by TwiN** [reply]
Please make another issue/discussion, since it's a different question. But to answer your question, I doubt `TZ` is taken into consideration; it probably uses UTC.
